make/modules/java.base/lib/CoreLibraries.gmk

```diff
@@ -48,7 +48,8 @@ $(eval $(call SetupNativeCompilation, BUILD_LIBFDLIBM, \
     CFLAGS := $(CFLAGS_JDKLIB) $(LIBFDLIBM_CFLAGS), \
     CFLAGS_windows_debug := -DLOGGING, \
     CFLAGS_aix := -qfloat=nomaf, \
-    DISABLED_WARNINGS_gcc := sign-compare misleading-indentation array-bounds, \
+    DISABLED_WARNINGS_gcc := sign-compare misleading-indentation \
+        array-bounds maybe-uninitialized, \
```

Comment:
Something like this would be better:
```
#pragma GCC diagnostic push
#pragma GCC diagnostic ignored "-Wmaybe-uninitialized"
    write(foo, bar, baz);
#pragma GCC diagnostic pop

```

Reply:
Thanks for your review.

Your mentioned solution should work and we can also enable the pragma only for aarch64 + gcc>=12 condition.
I considered such a solution when preparing this patch.

But I personally prefer to suppress the waring in the makefile, mainly because fdlibm is 3rd party library and we'd better not changing the source code. I think it is a long-standing policy.

Reply:
Yes, we should not modify 3rd party code.

What you would need is [JDK-8294281](https://bugs.openjdk.org/browse/JDK-8294281). If you can hold back this patch for a little while, you'll be able to utilize it instead.

Reply:
Thanks for your info @magicus.

Updated the code using [JDK-8294281](https://bugs.openjdk.org/browse/JDK-8294281) in the latest commit.
Would you mind taking another look? Thanks! @theRealAph  @magicus